moto/glue/models.py

```diff
@@ -136,9 +137,13 @@ def get_table(self, database_name, table_name):
         except KeyError:
             raise TableNotFoundException(table_name)
 
-    def get_tables(self, database_name):
+    def get_tables(self, database_name, expression=""):
         database = self.get_database(database_name)
-        return [table for table_name, table in database.tables.items()]
+        return [
+            table
+            for table_name, table in database.tables.items()
+            if bool(re.match(expression, table_name))
```

Comment:
With the earlier change, the expression will sometimes be None. So we should make sure that we handle that appropriately.

Also: I don't think the `bool()` conversion is necessary here. `re.match()` will return a None-object if there is no match, so we can test against that